controller/src/main/java/io/pravega/controller/store/stream/ZkInt96Counter.java

```diff
@@ -34,91 +30,16 @@
  * Once it exhausts the range, it refreshes the range by contacting zookeeper and repeating the steps described above. 
  */
 @Slf4j
-public class ZkInt96Counter {
-    /**
-     * This constant defines the size of the block of counter values that will be used by this controller instance.
-     * The controller will try to get current counter value from zookeeper. It then tries to update the value in store
-     * by incrementing it by COUNTER_RANGE. If it is able to update the new value successfully, then this controller
-     * can safely use the block `previous-value-in-store + 1` to `previous-value-in-store + COUNTER_RANGE` No other controller
-     * will use this range for transaction id generation as it will be unique assigned to current controller.
-     * If controller crashes, all unused values go to waste. In worst case we may lose COUNTER_RANGE worth of values everytime
-     * a controller crashes.
-     */
-    @VisibleForTesting
-    static final int COUNTER_RANGE = 10000;
-    @VisibleForTesting
-    static final String COUNTER_PATH = "/counter";
-
-    private final Object lock;
-    @GuardedBy("lock")
-    private final AtomicInt96 limit;
-    @GuardedBy("lock")
-    private final AtomicInt96 counter;
-    @GuardedBy("lock")
-    private CompletableFuture<Void> refreshFutureRef;
-    private ZKStoreHelper storeHelper;
+class ZkInt96Counter extends AbstractInt96Counter {
```

Comment:
The `Int90Counter` interface defined what it did in terms of ZooKeeper implying that no other implementations were possible or make sense. Either that is incorrect and needs to be fixed, or this class should be renamed to `Int96CounterImpl` because it is the one and only possible implementation.

Reply:
Updated the Javadoc for Int96Counter interface.